app/controllers/admin/routes.py

```diff
@@ -157,6 +159,76 @@ def rsvpLogDisplay(eventId):
     else:
         abort(403)
 
+@admin_bp.route('/event/<eventId>/renew', methods=['POST'])
+def renewEvent(eventId):
+    try: 
+        formData=request.form
+        try:
+            assert formData['timeStart'] < formData['timeEnd']
+        except AssertionError:
+            flash("End time must be after start time", 'warning')
+            return redirect(url_for('admin.eventDisplay', eventId = eventId))
+        
+        try:
+            if formData.get('dateEnd'):
+                assert formData['dateStart'] < formData['dateEnd']
+        except AssertionError:
+            flash("End date must be after start date", 'warning')
+            return redirect(url_for('admin.eventDisplay', eventId = eventId))
+
+
+        priorEvent = Event.get_by_id(eventId)
+        rsvpInfo = list(EventRsvp.select().where(EventRsvp.event == priorEvent).execute())
+        participantInfo = list(EventParticipant.select().where(EventParticipant.event == priorEvent).execute())
+        newEvent = Event(
```

Comment:
We already have a function that creates new events and what you have above is basically a repeat. Is there a reason you are doing the new event creation this way instead of modifying the existing function?

Reply:
Changed route to use the existing function.